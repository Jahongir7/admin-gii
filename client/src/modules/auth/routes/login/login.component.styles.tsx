import styled from 'styled-components';

export const FormContainer = styled.form`
  width: 300px;
  padding: 15px;
  border: 1px solid #dddddd;
  box-shadow: 0px 3px 5px #e2e2e2;

  .bp4-input-group {
    margin-bottom: 15px;
  }

  .bp4-button {
    margin-bottom: 10px;
  }
`;
