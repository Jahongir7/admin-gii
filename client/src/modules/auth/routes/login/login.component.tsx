import { hoc } from '@utils/hoc';
import { AuthLayout } from '@auth/components';
import { useLoginProps } from './login.props';
import { Redirect } from 'react-router-dom';
import { FormContainer } from './login.component.styles';
import { Button, InputGroup, Label } from '@blueprintjs/core';

export const Login = hoc(
  useLoginProps,
  ({
    profile,
    handleLockClick,
    showPassword,
    handleSubmit,
    register,
    handleLogin
  }) => {
    if (profile) {
      return <Redirect to='/' />;
    }
    const lockButton = (
      <Button
        icon={showPassword ? 'unlock' : 'lock'}
        minimal={true}
        onClick={handleLockClick}
      />
    );
    
    return (
      <AuthLayout>
        <FormContainer onSubmit={handleSubmit(handleLogin)}>
          <Label>
            Email
            <InputGroup
              {...register('email', { required: true })}
              placeholder='Email-ni kiriting'
              type='email'
            />
          </Label>
          <Label>
            Parol
            <InputGroup
              {...register('password', { required: true })}
              placeholder='Parolni kiriting'
              rightElement={lockButton}
              type={showPassword ? 'text' : 'password'}
            />
          </Label>
          <Button
            type='submit'
            rightIcon='lock'
            intent='success'
            text='Kirish'
          />
        </FormContainer>
      </AuthLayout>
    );
  },
  'Login'
);
