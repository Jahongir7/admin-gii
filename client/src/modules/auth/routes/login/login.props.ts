// import { useDispatch } from 'react-redux';
import { useHistory } from 'react-router-dom';
import { authActions, useAuthError, useAuthLoading } from '@auth/store';
import { useProfileSelector } from '@store/common';
import { useDispatch } from 'react-redux';
import { LoginFormState } from '@auth/model';
import { useState } from 'react';
import { useCustomForm } from '@hooks';

const initialValues: LoginFormState = {
  email: '',
  password: ''
};

export const useLoginProps = () => {
  const dispatch = useDispatch();
  const profile = useProfileSelector();
  const loading = useAuthLoading();
  const error = useAuthError();
  const [showPassword, setShowPassword] = useState(false);
  const {register, handleSubmit} = useCustomForm({
    defaultValues: initialValues
  });

  // const history = useHistory();

  const handleLogin = ({ email, password }: LoginFormState) => {
    console.log(email, password);
    dispatch(
      authActions.login({
        email,
        password
      })
    );
  };

  const handleLockClick = () => {
    setShowPassword(!showPassword);
  }
  
  return {
    loading,
    profile,
    error,
    handleLockClick,
    showPassword,
    handleSubmit,
    register,
    handleLogin
  };
};
