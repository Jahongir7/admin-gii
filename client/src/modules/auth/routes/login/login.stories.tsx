import { ComponentStory, ComponentMeta } from '@storybook/react';
import { Login } from './login.component';
import '@styles/index.css';
import { GlobalWrapper } from '@main';

export default {
  title: 'Auth/Login',
  component: Login,
} as ComponentMeta<typeof Login>

const Template: ComponentStory<typeof Login> = (args) => (
  <GlobalWrapper>
    <Login {...args} />
  </GlobalWrapper>
);

export const LoginInitial = Template.bind({});
LoginInitial.args = {
}