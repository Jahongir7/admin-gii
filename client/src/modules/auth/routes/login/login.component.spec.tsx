import { customRender } from '@test-utils';
import { Login } from './login.component';

test('Should render Login component', () => {
  const { asFragment } = customRender(<Login />);

  expect(asFragment).toMatchSnapshot();
});
