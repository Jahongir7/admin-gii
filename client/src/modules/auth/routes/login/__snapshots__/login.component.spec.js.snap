// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Login component 1`] = `[Function]`;
